### PR TITLE
Improve typespecs for Registry.start_link/1 and Registry.child_spec/1.

### PR DESCRIPTION
**Pull request by derekkraan**
I noticed that `child_spec/1` was missing a typespec, and "opts" seemed a little mysterious. This PR adds the missing typespec for `child_spec/1` and also cleans up the typespec for `start_link/1` (I'm not sure why that `when` was there, but it looks incorrect).

### CODE REVIEW

File: lib/elixir/lib/registry.ex [hunk 1]
```diff
@@ -209,6 +209,15 @@ defmodule Registry do
...
   @type spec :: [{match_pattern, guards, body}]
 
+  @typedoc "Options used for child_spec/1 and start_link/1"
+  @type options :: [
+          keys: keys,
+          name: registry,
+          partitions: pos_integer,
+          listeners: [atom],
+          meta: [{meta_key, meta_value}]
+        ]
```
Comment (whatyouhide) [inline lib/elixir/lib/registry.ex]: I think that the pattern that we usually use is:

```elixir
@type option :: {:keys, keys} | {:name, registry} | ...

@spec start_link([option]) :: ...
```

Wdyt?

Comment (whatyouhide) [reply lib/elixir/lib/registry.ex]: Also, I'd probably call this `start_option` instead. :)

Comment (derekkraan) [reply lib/elixir/lib/registry.ex]: @whatyouhide updated with your suggestions

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**